### PR TITLE
UG Bug Fix

### LINKED ISSUE #160: [PE-D] historyData.txt file
Similar to fridge data text file, can edit quantity in the text file to be negative or non-integers.
Expected an error.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: hussain1998/ped#8

### LINKED ISSUE #165: [PE-D] Possible UG Bugs
The UG looks good and well-formated.
It would be better to do some minor adjustment for the feature's sequence. For example, put Get help message: help at the beginning. It would be easier for user to refer your UG.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: Cocokkkk/ped#3

### LINKED ISSUE #176: [PE-D] Broken Link under Contents
The remove link doesn't bring you to anywhere.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: tzexern/ped#2

### LINKED ISSUE #178: [PE-D] Minor Grammar Mistake under Remove Feature
Under remove, notes for `QUANTITY_TO_REMOVE`, is `...must be lower to equal...` supposed to be `... lower or equal...`? 

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: tzexern/ped#4

### LINKED ISSUE #181: [PE-D] Insufficient information under `runnignlow`
Would it be better if this section mentioned that the `"...below a specified minimum limit"`'s minimum limit can be set by `setlimit` so that the users don't get confused?

Would it also be better if the user can view what their current `minimum limit` is?

EDIT:

Just realized that your `"...out of VALUE"` indicates the `minimum limit`. I think it would it be better to include that so that the users can find out the `minimum limit` that way?

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: tzexern/ped#7

### LINKED ISSUE #182: [PE-D] Extraneous parameters not handled
Extra parameters given for commands that requires no parameters can still execute as per usual.
This can be seen in other places as well:

and others.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: tzexern/ped#8

### LINKED ISSUE #186: [PE-D] UG Incorrect expected output under Remove feature
These 2 screenshots are the same output, due to your UG layout, I can't include it in one.

the list suggests that there are `1000 chicken` but after your remove command `remove chicken /qty 200`, the new quantity is `1300`.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: tzexern/ped#12

### LINKED ISSUE #189: [PE-D] UG Section Format is inconsistent
Here, the Introduction of FridgeFriend comes first and then the content page which includes all the different sections of the UG. A better way to format is maybe including the content first, then followed by the Introduction so that it is easier for the user to understand where each section is.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: vaiish371/ped#1

### LINKED ISSUE #195: [PE-D] Missing commands in help list
Commands "history", "history clear" and "list STORAGE_LOCATION_NAME", which are all stated in the User Guide, are not in the list of commands when "help" is entered into the program.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: 8kdesign/ped#1

### LINKED ISSUE #196: [PE-D] Missing command in User Guide
Command "clear" (not to be confused with "history clear") is missing from the User Guide.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: 8kdesign/ped#2

### PR DESCRIPTION
**Pull request by kwokyto**
- [x] #160  
- [x] #163 
- [x] #164 
- [x] #165 
- [ ] #174
- [x] #176 
- [x] #177 
- [x] #178 
- [x] #181 
- [x] #182 
- [x] #186 
- [x] #189 
- [x] #196
- [x] Add that we might add extra functionality to individual food classes in UG/DG "Future Work" section
- [x] #201 
- [x] future work —> modify to explain future ideas without mention of food cat
- [x] hide food category classes in sequence diagram
- [x] add warning in UG and DG for history data

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -271,20 +287,18 @@ Example of usage:
...
+6. BEVERAGE quantity: 0 out of 500
+7. COOKED_DISH quantity: 0 out of 500
+8. FROZEN quantity: 0 out of 500
+9. OTHER quantity: 0 out of 500
 ```
 
 ### Modify the minimum quantity limits: `setlimit`
 
-Changes the minimum quantity limit for a specific food category.
+Changes the minimum quantity limit for a specific food category. To set remove a limit for a category, set its limit to -1.
```
Comment (SimJJ96) [inline docs/UserGuide.md]: Would it be possible to change the limit to 0?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -255,12 +271,12 @@ Example of usage:
...
 >> expiring
 These are the food expiring in the next week:
-1. Food name: chicken wings, category: MEAT, expiry: 21-03-2021, stored in: MIDDLE_SHELF
+1. Food name: chicken wings, category: READY_TO_EAT, expiry: 17-04-2021, stored in: UPPER_SHELF, quantity: 500
 ```
 
 ### List categories with food running low: `runninglow`
 
-Displays a list of food categories which total quantity is below a specified minimum limit.
+Displays a list of food categories which total quantity is below a specified minimum limit. This command also displays the existing quantities along with its limits for each category. Note that these limits can be modified with the [`setlimit`](#modify-the-minimum-quantity-limits-setlimit) command.
```
Comment (SimJJ96) [inline docs/UserGuide.md:279]: Should we inform the user that if all of the minimum quantity if set to 0 would disable the running low command ?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -293,20 +307,33 @@ Example of usage:
...
 
->> runninglow
-You are running low on food in these categories:
-1. MEAT quantity: 250 out of 300
+>> setlimit ready_to_eat /qty 600 
+Okie dokie! The new minimum quantity for category 'READY_TO_EAT' is 600
 
->> setlimit meat /qty 200
-Okie dokie! The new minimum quantity for category 'MEAT' is 200
+>> setlimit vegetable /qty -1
```
Comment (SimJJ96) [inline docs/UserGuide.md]: Would you help me change to the newer version of setlimit vegetable /qty 0 ?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -293,20 +307,33 @@ Example of usage:
...
+Okie dokie! The new minimum quantity for category 'VEGETABLE' is -1
```
Comment (SimJJ96) [inline docs/UserGuide.md]: Similarly here would be "Okie dokie! The new minimum quantity for category 'VEGETABLE' is 0"

**Review by SimJJ96 — APPROVED** [review]
Could update the minor changes. Other than that LGTM

File: docs/UserGuide.md (hunk 1 above)
Comment (kwokyto) [reply docs/UserGuide.md]: modified

File: docs/UserGuide.md (hunk 3 above)
Comment (kwokyto) [reply docs/UserGuide.md]: modified

File: docs/UserGuide.md (hunk 4 above)
Comment (kwokyto) [reply docs/UserGuide.md]: modified

File: docs/UserGuide.md (hunk 2 above)
Comment (kwokyto) [reply docs/UserGuide.md:279]: rejected, error message on your side will change

**Review by Vinci-Hu — APPROVED** [review]
LGTM. Good practice to make the MinQuantity class.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -148,52 +149,22 @@ The `Exceptions` component currently consists of eleven (11) custom Exceptions,
...
-* **`StorageLoadingException`**: Is thrown when an error occurred during loading of the saved data.
-  * Error Message: `"There was an error loading the data for FridgeFriend!"`
-* **`StorageSavingException`**: Is thrown when an error occurred during saving of the current data.
-  * Error Message: `"There was an error saving the data for FridgeFriend!"`
+* Invalid User Input
+  * Thrown when a raw user input cannot be parsed into a valid value
+  * Includes: `EmptyDescriptionException`, `InvalidDateException`, `InvalidInputException`, `InvalidQuantityException`
+* Food Errors
+  * Thrown when a parameter for a food item is invalid
+  * Includes: `FoodNameNotFoundException`, `InvalidFoodCategoryException`, `RepetitiveFoodIdentifierException`
```
Comment (Leeyp) [inline docs/DeveloperGuide.md]: We are missing InvalidFoodStorageLocation Exception and the newest Minimum Quantity exception

**Review by Leeyp — APPROVED** [review]
Beautiful, excellent work! One reminder to update the Exceptions section because it appears that new exceptions have been added to the application.

Approved

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (kwokyto) [reply docs/DeveloperGuide.md]: minquantity will be done by @SimJJ96 
edited for for location

**Pull request merged**